client/src/controllers/DialogController.ts

```diff
@@ -34,13 +36,36 @@ export class DialogController extends Controller<HTMLElement> {
     this.dialog
       .on('show', () => {
         if (!isFloating) document.documentElement.style.overflowY = 'hidden';
-        this.dispatch('shown', { detail });
+        this.dispatch('shown', { detail, cancelable: false });
+        if (this.notifyTargets && Array.isArray(this.notifyTargets)) {
```

Comment:
There is no need for this defensive checking here, the good thing about Stimulus is that  `fooTargets` will always be an array, whether or not any targets are used.


See https://stimulus.hotwired.dev/reference/targets#properties

This means that we can safely call `this.notifyTargets.forEach(...` and be confident it will always work.

Reply:
About this, the pre-existing tests in the DialogController.test.js failed without it. That's actually the reason I put the defensive checks in the first place. They failed with a `TypeError: Cannot read properties of undefined (reading 'forEach')`

Reply:
Okay, I think I've gotten this part to work without the defensive checks.